{{cookiecutter.project_slug}}/frontend/src/store/modules/main.ts

```diff
@@ -0,0 +1,253 @@
+import { api } from "@/api";
+import { getLocalToken, removeLocalToken, saveLocalToken } from "@/utils";
+import router from "@/router";
+import { AxiosError } from "axios";
+import { VuexModule, Module, Mutation, Action } from "vuex-module-decorators";
+import { IUserProfile, IUserProfileUpdate, IAppNotification } from "@/interfaces";
+import { UNAUTHORIZED } from "http-status-codes";
+
+@Module({ name: "main" })
+export default class MainModule extends VuexModule {
+  token = "";
+  isLoggedIn: boolean | null = null;
+  logInError = false;
+  userProfile: IUserProfile | null = null;
+  dashboardMiniDrawer = false;
+  dashboardShowDrawer = true;
+  notifications: IAppNotification[] = [];
+
+  get hasAdminAccess() {
+    return (
+      this.isLoggedIn && this.userProfile?.is_superuser && this.userProfile?.is_active
+    );
+  }
+
+  get firstNotification() {
+    return this.notifications.length > 0 && this.notifications[0];
+  }
+
+  @Mutation
+  setToken(payload: string) {
+    this.token = payload;
+  }
+
+  @Mutation
+  setLoggedIn(payload: boolean) {
+    this.isLoggedIn = payload;
+  }
+
+  @Mutation
+  setLogInError(payload: boolean) {
+    this.logInError = payload;
+  }
+
+  @Mutation
+  setUserProfile(payload: IUserProfile) {
```

Comment:
Fix errorwhen trying to set user profile to `null`.

```suggestion
  setUserProfile(payload: IUserProfile | null) {
```